In [1]:
import re
import os
import time
import datetime
import random
import requests
import numpy as np
import pandas as pd
import ws_func
from datetime import datetime
from os import system   
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

## Scrape CNN Articles

In [2]:
# Scrape CNN articles and save to cnn.tsv (tab separated to avoid comma issues with text body)
ws_func.scrape_cnn(start=(2019, 1, 1), stop=(2020, 1, 1), ret_csv=True, csv='cnn.tsv', ret_df=False)

All URLS in date range scraped
Number of article URLs: 221

Number of Articles Scraped: 220

Number of Articles w/o Text: 1

File cnn.tsv Created


### Notes on CNN Articles
- Total number of articles: 221
- Date range of scrape: 2019 Jan 1 to 2019 Dec 31 (future scrapes from 2019 Mar 15 to 2019 Dec 31)
- Dropped (within the script) all 'live coverage' and 'style' section articles
- Articles with text: 220
- Articles w/o text: 1
- Article without text is summary article (to be dropped)

In [3]:
df = pd.read_csv('cnn.tsv', sep='\t', index_col=0)

In [4]:
df.loc[df['body'] == 'No text']

url        date  \
index                                                                  
205    https://www.cnn.com/2019/06/11/asia/hong-kong-...  2019-06-11   

          headline     byline     body source  
index                                          
205    No headline  No byline  No text    CNN

In [7]:
df.tail(20)

url        date  \
index                                                                  
201    https://www.cnn.com/2019/06/13/asia/china-hong...  2019-06-13   
202    https://www.cnn.com/2019/06/13/asia/hong-kong-...  2019-06-13   
203    https://www.cnn.com/2019/06/12/asia/hong-kong-...  2019-06-12   
204    https://www.cnn.com/2019/06/12/asia/hk-protest...  2019-06-12   
205    https://www.cnn.com/2019/06/11/asia/hong-kong-...  2019-06-11   
206    https://www.cnn.com/2019/06/11/asia/hong-kong-...  2019-06-11   
207    https://www.cnn.com/2019/06/08/asia/hong-kong-...  2019-06-08   
208    https://www.cnn.com/2019/06/10/asia/hong-kong-...  2019-06-10   
209    https://www.cnn.com/2019/06/03/asia/tiananmen-...  2019-06-03   
210    https://www.cnn.com/2019/06/02/asia/tiananmen-...  2019-06-02   
211    https://www.cnn.com/2019/06/03/asia/china-tian...  2019-06-03   
212    https://www.cnn.com/2019/04/18/asia/taiwan-ter...  2019-04-18   
213    https://www.cnn.com/2019/04/08/asia/hong-kong-...  2019-04-08   
214    https://www.cnn.com/2019/04/01/asia/philippine...  2019-04-01   
215    https://www.cnn.com/2019/03/04/asia/china-mili...  2019-03-04   
216    https://www.cnn.com/2019/02/21/africa/zambia-h...  2019-02-21   
217    https://www.cnn.com/2019/01/17/africa/internet...  2019-01-17   
218    https://www.cnn.com/2019/01/11/asia/us-xinjian...  2019-01-11   
219    https://www.cnn.com/2019/01/04/asia/china-inte...  2019-01-04   
220    https://www.cnn.com/2019/01/05/asia/xi-jinping...  2019-01-05   

                                                headline  \
index                                                      
201    Beijing walks a thin line as Hong Kong erupts ...   
202    How did one of the safest cities in the world ...   
203          The return of Hong Kong's umbrella movement   
204    Hong Kong police declare China extradition pro...   
205                                          No headline   
206    Does the extradition law really spell the deat...   
207    More than 1 million protest in Hong Kong, orga...   
208    Hong Kong's protest movement was on life suppo...   
209    World marks 30 years since Tiananmen massacre ...   
210    Tiananmen Square massacre: How Beijing turned ...   
211    Tiananmen massacre a 'vaccination' against tur...   
212    A goddess inspired a Taiwanese billionaire to ...   
213    Leaders of Hong Kong's Umbrella Movement found...   
214    Philippines protests hundreds of Chinese ships...   
215    China's military is going from strength to str...   
216    A long-disputed hippo cull will begin in May i...   
217    Internet shutdowns aren't just Africa's proble...   
218    US clothing company drops Chinese supplier ove...   
219    China's censors face a major test in 2019. But...   
220    China's Xi Jinping begins his most important y...   

                                                  byline  \
index                                                      
201       Analysis by Ben Westcott and Steven Jiang, CNN   
202                     Analysis by James Griffiths, CNN   
203                              By James Griffiths, CNN   
204                          By Julia Hollingsworth, CNN   
205                                            No byline   
206                     Analysis by James Griffiths, CNN   
207    By James Griffiths, Eric Cheung and Chermaine ...   
208                     Analysis by James Griffiths, CNN   
209                              By James Griffiths, CNN   
210                                 By Ben Westcott, CNN   
211                                 By Ben Westcott, CNN   
212                              By James Griffiths, CNN   
213              By Eric Cheung and James Griffiths, CNN   
214                                  By Brad Lendon, CNN   
215                                 By Ben Westcott, CNN   
216                               By Bukola Adebayo, CNN   
217                     Analysis by James Griffiths, CNN   
218    

In [ ]:
driver = webdriver.Chrome()
driver.get('https://www.cnn.com/search?q=Hong%20Kong%20Protest&size=10&type=article')